reactor-core/src/main/java/reactor/core/Exceptions.java

```diff
@@ -586,11 +585,56 @@ public synchronized Throwable fillInStackTrace() {
 		}
 	}
 
-	static final class ReactorRejectedExecutionException extends RejectedExecutionException {
+	static class ReactorRejectedExecutionException extends RejectedExecutionException {
 
 		ReactorRejectedExecutionException(String message, Throwable cause) {
 			super(message, cause);
 		}
+
+		ReactorRejectedExecutionException(String message) {
+			super(message);
+		}
+	}
+
+	/**
+	 * A {@link ReactorRejectedExecutionException} that is tailored for usage as a static final
+	 * field. It avoids {@link ClassLoader}-related leaks by bypassing stacktrace filling.
+	 */
+	static final class StaticRejectedExecutionException extends ReactorRejectedExecutionException {
+
+		StaticRejectedExecutionException(String message, Throwable cause) {
+			super(message, cause);
+		}
+
+		StaticRejectedExecutionException(String message) {
+			super(message);
+		}
+
+		@Override
+		public synchronized Throwable fillInStackTrace() {
+			return this;
+		}
+	}
+
+	/**
+	 * A general-purpose {@link Throwable} that is suitable for usage as a static final
+	 * field. It avoids {@link ClassLoader}-related leaks by bypassing stacktrace filling.
+	 * Exception {{@link Exception#addSuppressed(Throwable)} suppression} is also disabled.
+	 */
+	//see https://github.com/reactor/reactor-core/pull/1872
+	static final class StaticThrowable extends Error {
```

Comment:
`StaticThrowable` does not override `fillInStackTrace`

Reply:
The constructor writableStackTrace is used for that case. Is you use that, fillStackTrace is not used.

Reply:
TIL! 👍